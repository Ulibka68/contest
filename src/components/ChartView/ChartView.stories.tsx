import React, { useState } from "react";

import { ChartGrid } from "@/components/ChartGrid";
import { ChartView, IChartView } from "@/components/ChartView";
import { RangeSelector } from "@/components/RangeSelector";
import { Toggler, TogglerGroup } from "@/components/Toggler";
import { action } from "@storybook/addon-actions";

export default {
  title: "Example/AppComposition",
  component: ChartView,
};

const xAxisSamples = ["10", "11", "12", "13", "14", "15", "18", "16", "17"];
const area: IChartView = { width: 800, height: 400 };

const Template = () => {
  const [position, setPosition] = useState({ left: 0, right: 0 });
  return (
    <ChartView area={area}>
      <ChartGrid xAxis={xAxisSamples} area={area} />
      <RangeSelector position={position} setPosition={setPosition} />
      <TogglerGroup>
        <Toggler
          onClick={(status) => action(`status :${status.toString()}`)}
          label="mint"
          color="rgb(158, 212, 72)"
          isActive
        />
        <Toggler
          onClick={(status) => action(`status :${status.toString()}`)}
          label="strawberry"
          color="rgb(254, 60, 48)"
        />
      </TogglerGroup>
    </ChartView>
  );
};
export const Primary = Template.bind({});
