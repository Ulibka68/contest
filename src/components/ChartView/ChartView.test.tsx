import React from "react";
import { shallow } from "enzyme";
import { ChartView } from "./ChartView";

describe("ChartLine render", () => {
  const area = { width: 800, height: 400 };
  it("is ChartLine with stable date to match snapshot", () => {
    const wrapper = shallow(<ChartView area={area}>Test string</ChartView>);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
