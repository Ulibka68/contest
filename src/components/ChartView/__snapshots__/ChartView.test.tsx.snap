// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartLine render is ChartLine with stable date to match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-1b3grcd",
      "style": "width:800px",
    },
    "children": Array [
      Object {
        "data": "Test string",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
